kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1288 @@
+            }
+
+            auto& r_constraints_array = rModelPart.MasterSlaveConstraints();
+            const int number_of_constraints = static_cast<int>(r_constraints_array.size());
+            #pragma omp for  schedule(guided, 512)
+            for (int i = 0; i < number_of_constraints; ++i) {
+                auto it_const = r_constraints_array.begin() + i;
+
+                // Gets list of Dof involved on every element
+                it_const->GetDofList(dof_list, auxiliar_dof_list, r_current_process_info);
+                dofs_tmp_set.insert(dof_list.begin(), dof_list.end());
+                dofs_tmp_set.insert(auxiliar_dof_list.begin(), auxiliar_dof_list.end());
+            }
+
+            #pragma omp critical
+            {
+                dof_global_set.insert(dofs_tmp_set.begin(), dofs_tmp_set.end());
+            }
+        }
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", ( this->GetEchoLevel() > 2)) << "Initializing ordered array filling\n" << std::endl;
+
+        dof_temp_all.reserve(dof_global_set.size());
+        for (auto& dof : dof_global_set) {
+            dof_temp_all.push_back( dof.get() );
+        }
+        dof_temp_all.Sort();
+        BaseType::mDofSet = dof_temp_all;
+
+        // Repeating the loop for solvable DoF (in serial)
+        set_type dofs_aux_list_solvable;
+        // Gets the array of elements from the modeler
+        ElementsArrayType& r_elements_array = rModelPart.Elements();
+        const int number_of_elements = static_cast<int>(r_elements_array.size());
+        #pragma omp for schedule(guided, 512) nowait
+        for (int i = 0; i < number_of_elements; ++i) {
+            auto it_elem = r_elements_array.begin() + i;
+
+            // Gets list of Dof involved on every element
+            pScheme->GetElementalDofList(*(it_elem.base()), dof_list, r_current_process_info);
+            dofs_aux_list_solvable.insert(dof_list.begin(), dof_list.end());
+        }
+
+        // Gets the array of conditions from the modeler
+        ConditionsArrayType& r_conditions_array = rModelPart.Conditions();
+        const int number_of_conditions = static_cast<int>(r_conditions_array.size());
+        #pragma omp for  schedule(guided, 512)
+        for (int i = 0; i < number_of_conditions; ++i) {
+            auto it_cond = r_conditions_array.begin() + i;
+
+            // Gets list of Dof involved on every element
+            pScheme->GetConditionDofList(*(it_cond.base()), dof_list, r_current_process_info);
+            dofs_aux_list_solvable.insert(dof_list.begin(), dof_list.end());
+        }
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", ( this->GetEchoLevel() > 2)) << "Initializing constraints loop" << std::endl;
+
+        // We don't do a loop in OMP because erase is not a parallel threadsafe operation
+        auto& r_constraints_array = rModelPart.MasterSlaveConstraints();
```

Comment:
why don't you do this in the parallel loop above? instead of erasing you just add to a different list...

Reply:
True